packages/mobile/src/components/premium-track-purchase-drawer/PremiumTrackPurchaseDrawer.tsx

```diff
@@ -432,10 +432,17 @@ export const PremiumTrackPurchaseDrawer = () => {
 
   const isLoading = statusIsNotFinalized(trackStatus)
 
-  const isValidTrack = track && isTrackPurchaseable(track)
-  const price = isValidTrack
-    ? track?.stream_conditions?.usdc_purchase?.price
-    : 0
+  const isValidStreamGatedTrack = track && isTrackStreamPurchaseable(track)
+  const isValidDownloadGatedTrack = track && isTrackDownloadPurchaseable(track)
+  const isValidTrack = isValidDownloadGatedTrack || isValidStreamGatedTrack
+
+  const purchaseConditions = isValidStreamGatedTrack
+    ? track.stream_conditions
+    : isValidDownloadGatedTrack
+    ? track.download_conditions
+    : ({} as USDCPurchaseConditions)
+  const { price } = purchaseConditions.usdc_purchase
```

Comment:
I'd generally prefer we avoid things like `{} as USDCPurchaseConditions` since it removes our type safety. If I understand correctly, in the case where it's not a valid track (has neither type of conditions object), we're going to assign `{}` to purchaseConditions and then try to read `purchaseConditions.usdc_purchase.price`, which would throw an Error.

Would you mind restructuring this a bit so we can rely on the type safety?